digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/NumericalLimitFiller.scala

```diff
@@ -162,6 +162,160 @@ object NumericalLimitFiller {
     (segments ++ generated, changeSet)
   }
 
+  private def combine(roadLink: RoadLink, segments: Seq[PersistedLinearAsset], changeSet: ChangeSet): (Seq[PersistedLinearAsset], ChangeSet) = {
```

Comment:
This seems to be the same code that speed limits use. Is is possible to put it somewhere where both could use it, so there wouldn't be duplicate code? Or are there many differences?